jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
I think you're missing the point of a somewhat contrived example :) However deliberate, it shouldn't be too hard to imagine other use cases that fit the pattern of _do this thing (that returns an option) `andThen` do this other thing (that also returns an option) with it_. It's certainly a more imperative viewpoint, but most devs do actually come from the imperative world. `flatMap` on the other hand is incomprehensible to anyone who hasn't already memorized its function, although it does have the nice property of being fairly easy to remember once you do learn of it.